onnxruntime/core/session/ep_api.cc

```diff
@@ -176,6 +176,31 @@ ORT_API(uint32_t, MemoryDevice_GetDeviceId, _In_ const OrtMemoryDevice* memory_d
   return memory_device->Id();
 }
 
+ORT_API(const OrtSyncStreamImpl*, SyncStream_GetImpl, _In_ const OrtSyncStream* ort_stream) {
```

Comment:
we should ensure that no exceptions will be thrown from API functions.

Reply:
explicit try/catch at the level of the API function would work. this is probably the more maintainable approach. do you mean "heavy" in the sense of code to write or performance overhead? we already have macros which do this for OrtStatus APIs. otherwise, we need to ensure that code and other function calls in the API functions will not throw exceptions.

Reply:
Resolving purely to checkin the PR as any change along these lines will be a separate PR.